### PR TITLE
Periodic reporting of device information to a map via MQTT

### PR DESCRIPTION
**Pull request by GUVWAF**
This adds an easy way to periodically report all kinds of information about the device (long/short name, position with precision, firmware version, LoRa settings, number of local online nodes, etc.) to a map via MQTT only, regardless of any channel and is thus unencrypted. It will be published to a new `/2/map/` topic as a protobuf in a `ServiceEnvelope` with `MeshPacket` portnum `MAP_REPORT_APP` and can be decoded using the new `MapReport` message. 
If map reporting is enabled, the publish interval (default is 15 min.) and position precision (default is full precision) can be configured.

Implements #3336.

### CODE REVIEW

File: src/mqtt/MQTT.cpp [hunk 1]
```diff
@@ -77,8 +77,7 @@ void MQTT::onReceive(char *topic, byte *payload, size_t length)
                         if (jsonPayloadStr.length() <= sizeof(p->decoded.payload.bytes)) {
                             memcpy(p->decoded.payload.bytes, jsonPayloadStr.c_str(), jsonPayloadStr.length());
                             p->decoded.payload.size = jsonPayloadStr.length();
-                            meshtastic_MeshPacket *packet = packetPool.allocCopy(*p);
-                            service.sendToMesh(packet, RX_SRC_LOCAL);
+                            service.sendToMesh(p, RX_SRC_LOCAL);
```
Comment (caveman99) [inline src/mqtt/MQTT.cpp:80]: is there a reason why you removed the alloc?

**Review by caveman99 — CHANGES_REQUESTED** [review]
I updated protos for your convenience :-) Please see comment.

File: src/mqtt/MQTT.cpp (hunk 1 above)
Comment (GUVWAF) [reply src/mqtt/MQTT.cpp:80]: Oh, yeah that's actually unrelated to this pull, but I just stumbled upon it. I believe this leads to a memory leak as `p` wouldn't be freed. I believe the copy is unnecessary here; the "sendtext" part also doesn't use it.

Comment (caveman99) [reply src/mqtt/MQTT.cpp:80]: We can try, sounds plausible ... (how does telemetry handle this? there was a leak in there too...)

Comment (GUVWAF) [reply src/mqtt/MQTT.cpp:80]: At least currently it only allocates a `MeshPacket` once and uses that in `sendToMesh()`, as it should be:
https://github.com/meshtastic/firmware/blob/d1a25947e302cff9c0f1cb9b636569b67960fa9f/src/modules/Telemetry/DeviceTelemetry.cpp#L91

Comment (caveman99) [reply src/mqtt/MQTT.cpp:80]: sounds about right, telemetry was one of the sources of memory leaks... one more down :-)

**Review by caveman99 — APPROVED** [review]

**Pull request merged**